### PR TITLE
Pairing button, closes #52

### LINKED ISSUE #52: The + button shows a menu with Bluetooth, NFC and QR
[//]: # "Check if the title is descriptive!"
- Parent issues: -
- Relevant issues: -
- Estimated effort: 3

* * *

## User story
As a user I want the + button to show new buttons to add a new public key over Bluetooth, NFC and QR and I want the + button to hide the new buttons when it's clicked while the new buttons are displayed.

## What
An intuitive, user-friendly and clean way to make it possible to add new public keys using various connection mechanisms

## Why
This makes the interface look like a lot "cleaner" than when all buttons are visible initially

## How
To be found out...

## Definition of done
When the + button is clicked 3 new buttons appear above the + button. One contains a bluetooth logo, one contains an NFC logo and one contains a camera logo. When the + button is clicked again the new buttons disappear.

## Notes
-

### PR DESCRIPTION
**Pull request by jverbraeken**
- Relevant Issues: #52 
- Related Pull Requests: ...

* * *

## What
This PR adds a button to the right bottom corner of the MainActivity. When clicked, a menu opens with 3 buttons: bluetooth, nfc and qr.

## Why
This PR is needed for this project because the user must be able to choose between the various pairing methods when we offer these.

## How
This PR can be tested by clicking on the pairing button.

## Alternative implementation
I checked out quite some libraries and I chose Clang's library because it was quite flexible and easy-to-use.

## Notes
This PR was closed earlier for some reason, so this is merely a copy from the old pr

### CODE REVIEW

File: app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java [hunk 1]
```diff
@@ -34,7 +32,7 @@
 /**
  * The main activity class that shows a list of all people with their public keys
  */
-public final class MainActivity extends AppCompatActivity {
+public final class MainActivity extends Activity {
```
Comment (TheBonheurs) [inline app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java]: Why this switch? Weren't you the one saying AppCompatActivity  was better?

File: app/src/main/java/com/nervousfish/nervousfish/activities/ContactActivity.java [hunk 2]
```diff
@@ -31,7 +31,7 @@
 /**
  * The ContactActivity shows the contacts information and his public keys.
  */
-public final class ContactActivity extends AppCompatActivity {
+public final class ContactActivity extends Activity {
```
Comment (TheBonheurs) [inline app/src/main/java/com/nervousfish/nervousfish/activities/ContactActivity.java]: Same as below 👇

File: app/src/androidTest/java/com/nervousfish/nervousfish/test/startBluetoothConnectionSteps.java [hunk 3]
```diff
@@ -34,16 +36,41 @@ public void iHaveMainActivity() {
         assertNotNull(getActivity());
     }
 
-    @When("^I press the connect button$")
-    public void iPressTheConnectButton() {
-        onView(withId(R.id.fab)).perform(click());
+    @When("^I press the pairing button$")
```
Comment (TheBonheurs) [inline app/src/androidTest/java/com/nervousfish/nervousfish/test/startBluetoothConnectionSteps.java]: Good testing !

**Comment by TheBonheurs** [comment]
For the next time: it would be nice if you could add something about the focus of reveiwing in a pr that has this many changes otherwise 💯 / 💯

File: app/src/main/res/layout/activity_change_contact.xml [hunk 4]
```diff
@@ -9,7 +9,7 @@
     tools:context="com.nervousfish.nervousfish.activities.ContactActivity">
 
     <android.support.design.widget.AppBarLayout
-        android:id="@+id/app_bar"
+        android:id="@+id/app_bar_change_contact"
```
Comment (ericcornelissen) [inline app/src/main/res/layout/activity_change_contact.xml]: I like the name change but I would suggest to put the `change_contact` part at the beginning of the ID so all ID's related to the change contact part start the same

File: app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java [hunk 5]
```diff
@@ -96,6 +73,21 @@ public void onItemClick(final AdapterView<?> parent, final View view, final int
         LOGGER.info("MainActivity created");
     }
 
+    public void bluetoothButtonClicked(final View view) {
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java]: Naming a method in the past tense is very uncommon, maybe change this to `onBluetoothButtonClick` (similar approach for `nfcButtonClicked` and `qrButtonClicked`)

**Comment by jverbraeken** [comment]
@TheBonheurs 
> For the next time: it would be nice if you could add something about the focus of reveiwing in a pr that has this many changes otherwise 💯 / 💯

 No problem, will do that next time

File: app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java [hunk 6]
```diff
@@ -160,9 +160,9 @@ public void onStart() {
      */
     @Override
     public void onStop() {
-        this.serviceLocator.unregisterFromEventBus(this);
-        bluetoothHandler.stop();
+        LOGGER.info("Stopping from activity");
         super.onStop();
+        //bluetoothHandler.stop();
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/activities/BluetoothConnectionActivity.java]: Can you remove commented code?

**Comment by TheBonheurs** [comment]
You still have some pmd warnings 😢

**Pull request closed without merging**